lib/streamlit/delta_generator.py

```diff
@@ -797,17 +775,19 @@ def get_last_dg_added_to_context_stack() -> DeltaGenerator | None:
 
 def _prep_data_for_add_rows(
     data: Data,
-    delta_type: str,
-    add_rows_metadata: AddRowsMetadata,
-) -> tuple[Data, AddRowsMetadata]:
+    add_rows_metadata: AddRowsMetadata | None,
+) -> tuple[Data, AddRowsMetadata | None]:
     out_data: Data
 
-    # For some delta types we have to reshape the data structure
+    # For built-in chart commands we have to reshape the data structure
     # otherwise the input data and the actual data used
     # by vega_lite will be different, and it will throw an error.
-    if delta_type in ARROW_DELTA_TYPES_THAT_MELT_DATAFRAMES:
+    if add_rows_metadata:
```

Comment:
nit: if you turn the logic around, the function might be easier to digest because the current `else` branch is so small:

```python

if not add_rows_metadata:
  # When calling add_rows on st.table or st.dataframe we want styles to pass through.
  return type_util.convert_anything_to_df(data, allow_styler=True), add_rows_metadata
 
import pandas as pd
...
```

Reply:
Changed 👍